models/User.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * User Model
+ *
+ * Represents a registered user in the system.
+ * Handles identity, access, and relationships to projects, comments, etc.
+ */
+
+module.exports = (sequelize, Sequelize) => {
+  const { DataTypes } = Sequelize;
+
+  const User = sequelize.define(
+    "User",
+    {
+      id: {
+        type: DataTypes.UUID,
+        defaultValue: DataTypes.UUIDV4,
+        primaryKey: true,
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      username: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        unique: true,
+        validate: {
+          len: [3, 30],
+        },
+        set(value) {
+          this.setDataValue("username", value.toLowerCase().trim());
+        },
+      },
+      email: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        unique: true,
+        validate: {
+          isEmail: true,
+        },
+        set(value) {
+          this.setDataValue("email", value.toLowerCase().trim());
+        },
+      },
+      hashedPassword: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        comment: "Bcrypt-hashed password",
+      },
+      avatarUrl: {
+        type: DataTypes.STRING,
+        allowNull: true,
+        validate: { isUrl: true },
+        comment: "Optional user profile image URL",
+      },
+      bio: {
+        type: DataTypes.TEXT,
+        comment: "Optional short bio/about section",
+      },
+      roleId: {
+        type: DataTypes.INTEGER,
+        allowNull: true, // Allow role to be removed without deleting the user
+        defaultValue: 1, // Assuming 1 will be a default user role
+      },
+    },
+    {
+      timestamps: true,
+      tableName: "Users",
+      indexese: [{ fields: ["roleId"] }],
```

Comment:
Could index the emails to speed up logins
```suggestion
      indexes: [
        { fields: ["roleId"] },
        { fields: ["email"] },
      ],
```